packages/compute-baseline/src/browser-compat-data/supportStatements.ts

```diff
@@ -107,9 +90,91 @@ export class SupportStatement {
   }
 
-  get version_removed(): VersionValue {
-    // Strictness guarantee: unset version_removed returns false
-    return this.data?.version_removed || false;
+  get version_removed(): string | boolean | undefined {
+    const value = this.data?.version_removed;
+
+    // TODO: Report and fix upstream bug in BCD, then uncomment or drop the code
+    // below
+
+    // According to @mdn/browser-compat-data's schema, `version_removed` values
+    // should only be `true` or strings. In practice (and according to the
+    // exported types), this is not the case, because mirroring inserts `false`
+    // values.
+
+    // if (value === null || value === false) {
+    //   throw new Error(
+    //     "`version_added` should never be `null` or `false`. This is a bug, so please file an issue!",
+    //   );
+    // }
+    if (value === null) {
+      throw new Error(
+        "`version_added` should never be `null`. This is a bug, so please file an issue!",
+      );
+    }
+
+    return value;
+  }
+
+  /**
+   * Expand this support statement into a `Map` from `Release` objects to objects
+   * describing whether the release is supporting, unsupporting, or unknown.
+   */
+  toReleaseSupportMap(): ReleaseSupportMap {
+    if (this.browser === undefined) {
+      throw Error("This support statement's browser is unknown.");
+    }
+
+    if (this.version_added === false || this.version_removed === true) {
+      return new Map(
+        this.browser.releases.map((r) => [r, { supported: false }]),
+      );
+    }
+
+    if (this.version_added === null) {
+      return new Map(
+        this.browser.releases.map((r) => [r, { supported: null }]),
+      );
+    }
+
+    const result = new Map();
+
+    let start: Release;
+    let startRanged = false;
+    if (this.version_added === true) {
+      startRanged = true;
+      start = this.browser.current();
+    } else if (this.version_added.startsWith("≤")) {
+      startRanged = true;
+      start = this.browser.version(this.version_added.slice(1));
+    } else {
+      start = this.browser.version(this.version_added);
+    }
+
+    let end: Release | undefined;
+    if (this.version_removed) {
+      end = this.browser.version(this.version_removed);
+    }
+
+    const qualifications = statementToQualifications(this);
+    const isQualified = Boolean(Object.keys(qualifications).length);
+
+    for (const r of this.browser.releases) {
+      if (isQualified && r.inRange(start, end)) {
+        // Supported with qualifications
+        result.set(r, { supported: true, qualifications });
+      } else if (r.inRange(start, end)) {
+        // Supported without qualification
+        result.set(r, { supported: true });
+      } else if (startRanged && !r.inRange(start)) {
+        // Support unknown (before a ≤ version)
+        result.set(r, { supported: null });
+      } else {
+        // Unsupported (outside a hard range)
+        result.set(r, { supported: false });
+      }
```

Comment:
How about putting this into a method to get support for a specific release? That's cheaper when the caller doesn't want to know about all releases.